experiments/ClimaEarth/user_io/postprocessing.jl

```diff
@@ -0,0 +1,123 @@
+
+Conditionally plot AMIP diagnostics and call `common_postprocessing` to perform
+postprocessing tasks that are common to all simulation types.
+"""
+function postprocess_sim(mode_type::AMIP_mode, cs, postprocessing_vars)
+    (;
+        dir_paths,
+        plot_diagnostics,
+        use_coupler_diagnostics,
+        output_default_diagnostics,
+        t_end,
+        conservation_softfail,
+        atmos_output_dir,
+        amip_diags_handler,
+        comms_ctx,
+    ) = postprocessing_vars
+
+    if use_coupler_diagnostics
+        ## plot data that correspond to the model's last save_hdf5 call (i.e., last month)
+        @info "AMIP plots"
+
+        ## ClimaESM
+        include("user_io/diagnostics_plots.jl")
+
+        # define variable names and output directories for each diagnostic
+        amip_short_names_atmos = ["ta", "ua", "hus", "clw", "pr", "ts", "toa_fluxes_net"]
+        amip_short_names_coupler = ["F_turb_energy"]
+        output_dir_coupler = dir_paths.output
+
+        # Check if all output variables are available in the specified directories
+        make_diagnostics_plots(
+            atmos_output_dir,
+            dir_paths.artifacts,
+            short_names = amip_short_names_atmos,
+            output_prefix = "atmos_",
+        )
+        make_diagnostics_plots(
+            output_dir_coupler,
+            dir_paths.artifacts,
+            short_names = amip_short_names_coupler,
+            output_prefix = "coupler_",
+        )
+    end
+
+    # Check this because we only want monthly data for making plots
+    if t_end > 84600 * 31 * 3 && output_default_diagnostics
+        include("leaderboard/leaderboard.jl")
+        leaderboard_base_path = dir_paths.artifacts
+        compute_leaderboard(leaderboard_base_path, atmos_output_dir)
+        compute_pfull_leaderboard(leaderboard_base_path, atmos_output_dir)
+    end
+    common_postprocessing(cs, postprocessing_vars)
+end
+
+"""
+    common_postprocessing(cs, postprocessing_vars)
+
+Perfrom postprocessing common to all simulation types.
+"""
+function common_postprocessing(cs, postprocessing_vars)
+    (; plot_diagnostics, comms_ctx, dir_paths, amip_diags_handler, atmos_output_dir) = postprocessing_vars
+    if plot_diagnostics
+        @info "Plotting diagnostics"
+        include("user_io/diagnostics_plots.jl")
+        make_diagnostics_plots(atmos_output_dir, dir_paths.artifacts)
+    end
+
+    ## plot all model states and coupler fields (useful for debugging)
+    !CA.is_distributed(comms_ctx) && debug(cs, dir_paths.artifacts)
+
+    # if isinteractive() #hide
+    #     ## clean up for interactive runs, retain all output otherwise #hide
+    #     rm(dir_paths.output; recursive = true, force = true) #hide
+    # end #hide
+
+    ## close all AMIP diagnostics file writers
+    !isnothing(amip_diags_handler) && map(diag -> close(diag.output_writer), amip_diags_handler.scheduled_diagnostics)
```

Comment:
It would be good for this to be isolated to the AMIP postprocessing method. It needs to be called last (which is maybe why it's here), but we can accomplish both of those things if we call `common_postprocessing` at the start of each mode-specific postprocess method, and move this to the end of the AMIP method.